publication-commons/src/test/java/cucumber/permissions/file/FileAccessFeatures.java

```diff
@@ -58,6 +60,12 @@ public void theUserHaveTheRole(String userRole) {
         if (roles.contains(OTHER_CONTRIBUTORS)) {
             scenarioContext.addCurrentUserAndTopLevelAsContributor();
         }
+
+        if(roles.contains(EXTERNAL_CLIENT)) {
+            scenarioContext.setUserClientType(UserClientType.EXTERNAL);
+            scenarioContext.setPublisherId(scenarioContext.getCurrentUserInstance().getCustomerId());
```

Comment:
external client always has customerId? Or should it be null checked?

Reply:
it should as far as I know